### DISCUSSION TITLE
Getting issues while calling retire and discard APIs

### DISCUSSION THREAD

**Discussion by mahajanmahesh935** [Sunbird-Knowlg/Community#212] [Bugs]
I am reaching out to inform you about an issue we are currently experiencing with the Retire and discard API integration. Despite following the provided document, we have encountered difficulties in getting the Retire and discard API to function properly.

Here are the details of our attempts:

We have successfully onboarded and tested the following APIs:

Create Event
Update Event
Read Event
publish Event
However, when attempting to integrate the Retire and discard API, we encountered a 404 error response for both upstream URLs:

"{{ vm_learning_service_url }}/v4/retire"
"{{ knowledge_mw_service_url }}/v4/retire"
"{{ vm_learning_service_url }}/v4/discard"
"{{ knowledge_mw_service_url }}/v4/discard"
We have tried using both upstream URLs, but unfortunately, neither has resolved the issue.

Below is the code snippet we've implemented for reference:

- name: PrivateEventRetireAPI
  uris: "{{ private_event_prefix }}/v4/retire"
  upstream_url: "{{ vm_learning_service_url }}/event/v4/retire"
  strip_uri: true
  plugins:
  - name: jwt
  - name: cors
  - "{{ statsd_pulgin }}"
  - name: acl
    config.whitelist:
    - contentSuperAdmin
  - name: rate-limiting
    config.policy: local
    config.hour: "{{ medium_rate_limit_per_hour }}"
    config.limit_by: credential
  - name: request-size-limiting
    config.allowed_payload_size: "{{ small_request_size_limit }}"
  - name: opa-checks
    config.required: false
    config.enabled: false

- name: discardEvent
  uris: "{{ event_prefix }}/v4/discard"
  upstream_url: "{{ content_service_url }}/event/v4/discard"
  strip_uri: true
  plugins:
  - name: jwt
  - name: cors
  - "{{ statsd_pulgin }}"
  - name: acl
    config.whitelist:
    - contentAdmin
  - name: rate-limiting
    config.policy: local
    config.hour: "{{ medium_rate_limit_per_hour }}"
    config.limit_by: credential
  - name: request-size-limiting
    config.allowed_payload_size: "{{ medium_request_size_limit }}"
  - name: opa-checks
    config.required: false
    config.enabled: false

Could you please assist us in troubleshooting and resolving this issue with the Retire and discard API integration? Any guidance or support you can provide would be greatly appreciated.

Thank you for your attention to this matter.

**Comment by pallakartheekreddy** [comment]
@mahajanmahesh935 Please try to use `content_service_url` URL. Because Event and Event Set APIs are part of content-servcie.

**Reply by pallakartheekreddy** [reply]
@mahajanmahesh935 Pls let us know if your query is resolved

**Comment by mahajanmahesh935** [comment]
@pallakartheekreddy We are still facing the same issue we have tried changing upstream url as well config.whitelist: to creator but the issue is not resolved.

**Comment by mahajanmahesh935** [comment]
@pallakartheekreddy We are using release 4.10.0 of Sunbird-Knowlg, we found the end point for event retire API in `content-api/content-service/conf/routes` but we didn't found any function of retire API in controller can you please guide us how we can debug the retire API
https://github.com/Sunbird-Knowlg/knowledge-platform/blob/release-4.10.0_RC2/content-api/content-service/app/controllers/v4/EventController.scala
